tests/spread/rockcraft/plugin-maven/rockcraft.yaml

```diff
@@ -0,0 +1,32 @@
+name: plugin-maven
+base: bare
+build-base: ubuntu@24.04
+
+version: '0.1'
+summary: A rock that bundles a Maven project.
+description: A rock that bundles a Python project.
+license: GPL-3.0
+platforms:
+  amd64:
+
+parts:
+  maven-sample:
+    plugin: maven
+    source: sample
+    maven-parameters: [ "-B" ]
+    stage-packages:
+     - openjdk-21-jre-headless_core
+     - base-files_base
+    build-packages:
+      - maven
+      - openjdk-21-jdk-headless
+    # replace chiselled image modules with jlink output
+    # in order to ensure that only java.base module is present in
+    # runtime
+    override-stage: |
```

Comment:
how come this is done in `stage`? Can you do it in `build` instead?

Reply:
Notice 
```
    stage-packages:
     - openjdk-21-jre-headless_core
```
lib/modules have to be replaced after `openjdk-21-jre-headless_core` is staged. Hence this is done in stage.

Reply:
This will override jlink-generated module with the staged one:
```
spring-jl17      latest          56cb9c14956a   About an hour ago   348MB
spring           latest          b2ad2b9fe0e1   About an hour ago   186MB
```
Notice 348MB image ...

Reply:
But how can that be? The `stage-packages` are extracted into `CRAFT_PART_INSTALL`  *before*  the build step is executed

Reply:
Is there any chance of non-deterministic behaviour there? I will try to build with --shell-after and check the contents of `CRAFT_PART_INSTALL`

Reply:
no, this part should be pretty well-established; many plugins rely on the fact that stage-packages are already extracted when the build is executed.

Reply:
Ah, ok, this one is my fault - somehow in the image I've got a full install of openjdk-17-jre-headless and slices from openjdk-21-jre-headless.
I guess the issue was "not enough sleep".